### ISSUE TITLE
Support for Azure DevOps repos & Azure AD sign-in

### ISSUE THREAD

**Issue by empz** [gitpod-io/gitpod#5321]
A lot of companies are still using Azure DevOps, even though GitHub was acquired by Microsoft.

It would be nice to add support to add repos from an Azure DevOps organization. I supposed this would include adding SSO for Azure AD.

**Comment by kenzan-mryan** [comment]
We also have clients who can only use Azure Repositories, so I second he comments above.

**Comment by michael-bowen-sc** [comment]
We are heavily invested in Azure DevOps and would like to see it supported as well.

**Comment by sheldonhull** [comment]
I'm guessing I could compromise with this by having a github repo that defines just some scripts and gitpod config, inject in a secret for my access token and then clone inside the repo or as part of my bootstrap setup.

I don't think Azure DevOps integration is likely based on the future long-term compared to GitHub (most of the effort is now going to GitHub), but maybe a create work-around would give some options. 

I'm thinking an approach similar to [azdo-in-codespaces](https://github.com/codespaces-contrib/azdo-in-codespaces)

**Comment by ryanberger-az** [comment]
Hi,

We are also very interested in Azure DevOps integration. Right now, I am running a GitPod project out of a private GitHub with our team having access, but our main code repositories are within ADO. You can see how it is a pain in the butt to use a GitHub repo to build our GitPod environments, then commit code to a GitHub... then have to make those same changes into our Azure DevOps Repo's environment to push those code changes to Dev/Staging/Production.

Having direct integration with Azure DevOps via Azure AD auth would be incredible and would really take GitPod from being a proof of concept for us, to something we would not be able to live without.

Please let us know if this will soon be a part of the road map.

**Comment by satarr** [comment]
Highly needed feature. Please look into adding this functionality.

**Comment by monch1962** [comment]
Just another me-too request for Azure DevOps integration...

**Comment by dcb99** [comment]
My job uses Azure DevOps and we are looking to migrate to cloud workspaces, would be nice if gitlab had an integration.

**Comment by faustool** [comment]
> I don't think Azure DevOps integration is likely based on the future long-term compared to GitHub (most of the effort is now going to GitHub), but maybe a create work-around would give some options.

I don't think this is a valid concern for an Azure DevOps integration - https://www.linkedin.com/pulse/azure-devops-future-after-github-acquisition-microsoft-antonio-alvino/

**Comment by faustool** [comment]
> Having direct integration with Azure DevOps via Azure AD auth would be incredible and would really take GitPod from being a proof of concept for us, to something we would not be able to live without.

@ryanberger-az - I had the same issue and I've been working with a repository locally with two origins to push code to. It really sucks. I can't really promote it to do anything serious in my organization without this integration.

**Comment by aparedes** [comment]
This could be a great asset to have in any big company that uses azure and can’t move their code to GitHub.

**Comment by sheldonhull** [comment]
It’s also important to note the initial buzz was that Azure Devops was on a slow decline and would be phased out. I think the initial impression was inaccurate or the vision for it shifted as there is no real deprecation of the product and it’s used by many enterprises. This market is missed by Codespaces right now as well.

**Comment by shadowoftheknight** [comment]
I would like to see this feature as well in azure devops

**Comment by wolfgangihloff** [comment]
What is the suggested workaround to make gitpod work, setup replication for repo to github or bitbucket?

**Comment by DaveSmartEnt** [comment]
This prevents us from using gitpod, we are too deep into azure devops to consider moving to back to git and the "work around" isn't appealing.

**Comment by cpatrickalves** [comment]
Upvote! 
I am also using Azure Devops

**Comment by schmidjo82** [comment]
Upvote! We also using Azure DevOps. We are an Organization with more than 60 software developers

**Comment by SebastianSchuetze** [comment]
I am working at Deutsche Bahn and we have around 4000 users on Azure DevOps with currently over 500 people committing code on a regular basis.

Additionally Microsoft themselves stated that the investment will not likeley go away in the next years. Since last year the invested even more with GitHub Advanced Security integration in Azure DevOps. Azure DevOps itsel has a user base with over 3 Million world wide. 

I think an integration is feasable and not hard when it comes to authentication and discovery of repos.

**Comment by SebastianSchuetze** [comment]
Btw the authentication is working with Entra ID by sing OIDC Authentication.

https://www.gitpod.io/docs/configure/workspaces/oidc

**Comment by dooleydo** [comment]
What is the latest on this? I am looking at CDEs for my org, but also have a 3000+ userbase in ADO. No intention to move to GitHub Ent just yet, nor any desire to mirror.

How does this lack of integration affect the user journey?

**Comment by axonasif** [comment]
@dooleydo can you send us a message at contact@gitpod.io ?
